### ISSUE TITLE
CircuitBreaker counter doesn't reset counter on successful operation

### ISSUE THREAD

**Issue by Finity** [App-vNext/Polly#131]
Hey there, 

I could be wrong on this, but according to the code, we only reset the CircuitBreaker's internal counter when in an half-open state (which will only be reached once the counter reaches the threshold).  There's some confusion when reading the wiki, which states this:

"The above example will create a circuit-breaker which would break after two _consecutive_ exceptions of the handled type", however, if we have successes in between, we should never see the circuit trip (if I'm not mistaken).

```
public override void OnActionSuccess(Context context)
{
    using (TimedLock.Lock(_lock))
    {
        if (_circuitState == CircuitState.HalfOpen) { OnCircuitReset(context); }
    }
}
```

Should become:
```
public override void OnActionSuccess(Context context)
{
    using (TimedLock.Lock(_lock))
    {
        if (_circuitState == CircuitState.HalfOpen || _circuitState == CircuitState.Closed) { OnCircuitReset(context); }
    }
}
```

Please correct me if I'm wrong.

Thanks!

**Comment by reisenberger** [comment]
@Finity Thanks for this.  This will be fixed (in a slightly more optimised fix) in #130 (release v4.3.0), due out in the next few days.

**Comment by Finity** [comment]
No problem, thanks for the very quick fixes!
